### PR TITLE
Building near FOB will be 3 times slower than in the field

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Building near FOB will be 3 times slower than building in the field.

## Why It's Good For The Game

Engies are full fobiting, this is extremly detrimental to the game:
- They are not helping the push, so marines are getting shreded with no place to rest and heal
- They are slowing down the end of the game tremendously, because xeno do not have proper siege capabilities. This is not making marines win; this is not making people have fun; this is just a pain for everyone

Most, if not all are playing minecraft. This would be fine if that was not ruining the game for everyone else. This PR is made to incentivize engie to actually try to be usefull, rather than just playing on their own producing absolutly nothing

## Changelog
:cl:
balance: Building near FOB will be 3 times slower than in the field
/:cl:

### CODE REVIEW

**Comment by Pariah919** [comment]
You should really stop attemping to make people play the game in the way you want it to be played, as they'll just leave, but I doubt you really care anymore, as all of your actions show that you wish the game to be massively incentivized around the marines being forced to do an all out push or just alt f4, as there's no point in playing anymore.
Making FOBs and planning them out is fun for people, so why punish them if they enjoy it?

Stating minifobs are useful ignores the reality that they rarely last more than a moment most of the time, and if your first ideal idea is to jump to making it take longer to make a normal fob then people will just ignore that and do it anyway, your reasonings are just an excuse to nerf FOB making in the long term for no real conceviable reason.

**Comment by BraveMole** [comment]
> You should really stop attemping to make people play the game in the way you want it to be played, as they'll just leave, but I doubt you really care anymore, as all of your actions show that you wish the game to be massively incentivized around the marines being forced to do an all out push or just alt f4, as there's no point in playing anymore.
> Making FOBs and planning them out is fun for people, so why punish them if they enjoy it?
> 
> Stating minifobs are useful ignores the reality that they rarely last more than a moment most of the time, and if your first ideal idea is to jump to making it take longer to make a normal fob then people will just ignore that and do it anyway, your reasonings are just an excuse to nerf FOB making in the long term for no real conceviable reason.

So you are really defending the long as fuck sieges we have right now? Astounishing

**Comment by SpaceLoveSs13** [comment]
> You should really stop attemping to make people play the game in the way you want it to be played, as they'll just leave, but I doubt you really care anymore, as all of your actions show that you wish the game to be massively incentivized around the marines being forced to do an all out push or just alt f4, as there's no point in playing anymore.
> Making FOBs and planning them out is fun for people, so why punish them if they enjoy it?
> 
> Stating minifobs are useful ignores the reality that they rarely last more than a moment most of the time, and if your first ideal idea is to jump to making it take longer to make a normal fob then people will just ignore that and do it anyway, your reasonings are just an excuse to nerf FOB making in the long term for no real conceviable reason.

So is chaz gonna come and defend you and your cadehugging" too dude? FOB seiege is cancer and It was been 11 consecutive rounds that I have experienced fobbiting. Engineers making gigantic fob that is making xenoes either suicide or give up. Instead of posting snarky comments here, play the actual game. 11 rounda of fobbitting is a clear indication engi buff didn't fullfill t's intentions and conditions it was merged with.

**Comment by Windryn** [comment]
I do not think a reactionary PR submitted so soon is the best approach to this problem. 

This could very much eliminate the Marines agility to make mini FOBs

The most concerning thing however is the lack of ability to now patch a FOB, as now one breach to a cadeline with this update is much more of a death sentence that it previously would have been. Essentially boilers now can spam gas and then engineers have no hope to replace the cade during combat. 

Essentially this PR either makes a engie sit next to their cades to repair them as soon as they take damage, as to not lose them. or the more likely outcome, those who still bother to suffer to play marines engineer simply wont.

**Comment by BraveMole** [comment]
> I do not think a reactionary PR submitted so soon is the best approach to this problem.
> 
> This could very much eliminate the Marines ability to make mini FOBs
> 
> The most concerning thing however is the lack of ability to now patch a FOB, as now one breach to a cadeline with this update is much more of a death sentence that it previously would have been. Essentially boilers now can spam gas and then engineers have no hope to replace the cade during combat.
> 
> Essentially this PR either makes a engie sit next to their cades to repair them as soon as they take damage, as to not lose them. or the more likely outcome, those who still bother to suffer to play marine engineer simply wont.

This makes engie have the same building speed as before the buff. When fob sieging was already an issue.
And if engie cannot patch of FOB, good. FOB siege are not supposed to end up 5 hours.

**Comment by Windryn** [comment]
> This makes engie have the same building speed as before the buff. When fob sieging was already an issue.


So w're just returning to an issue we already had?

**Comment by Naaanii** [comment]
> I do not think a reactionary PR submitted so soon is the best approach to this problem.
> 
> This could very much eliminate the Marines ability to make mini FOBs
> 
> The most concerning thing however is the lack of ability to now patch a FOB, as now one breach to a cadeline with this update is much more of a death sentence that it previously would have been. Essentially boilers now can spam gas and then engineers have no hope to replace the cade during combat.
> 
> Essentially this PR either makes a engie sit next to their cades to repair them as soon as they take damage, as to not lose them. or the more likely outcome, those who still bother to suffer to play marine engineer simply wont.

The point of the PR is to make it easier for Xenos to breach and FOB defence harder, so once there is a breach, it becomes extremely difficult to fix with Xenos constantly pressuring. It's just slightly more forced gameplay trying to make engineers use their buffs in the intended manner.

**Comment by BraveMole** [comment]
> > This makes engie have the same building speed as before the buff. When fob sieging was already an issue.
> 
> So w're just returning to an issue we already had?

We're returning to an issue, that was way worse than what is now. Engie buffs just allowed worse fob siege, that all

**Comment by InterroLouis** [comment]
Does this affect the FOB drone?

**Comment by Tupinambis** [comment]
I think this artificial solution might not be the best solution and will probably confuse players. I hate Fobbiting to death but I think reworking the distress gamemode itself to discourage fobbiting is the way to go (by giving marines a timed objective they MUST push out to achieve or they lose)

File: code/game/objects/items/stacks/sandbags.dm [hunk 1]
```diff
@@ -106,6 +106,10 @@
 	user.visible_message("<span class='notice'>[user] starts assembling a sandbag barricade.</span>",
 	"<span class='notice'>You start assembling a sandbag barricade.</span>")
 	var/building_time = LERP(2 SECONDS, 1 SECONDS, user.skills.getPercent("construction", SKILL_ENGINEER_MASTER))
+	var/area/build_area = get_area(usr)
+	if(CHECK_BITFIELD(build_area.flags_area, NEAR_FOB))
+		building_time *= 3
+		to_chat(usr, "<span class ='warning'>The fuel residues of the space ship prevent us from using our power tool, we build slower!</span>")
```
Comment (SplinterGP) [inline code/game/objects/items/stacks/sandbags.dm:112]: Message span everytime you build

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -171,6 +171,10 @@
 			else
 				usr.visible_message("<span class='notice'>[usr] starts building \a [R.title].</span>",
 				"<span class='notice'>You start building \a [R.title]...</span>")
+			var/area/build_area = get_area(usr)
+			if(CHECK_BITFIELD(build_area.flags_area, NEAR_FOB))
+				building_time *= 3
+				to_chat(usr, "<span class ='warning'>The fuel residues of the space ship prevent us from using our power tool, we build slower!</span>")
```
Comment (SplinterGP) [inline code/game/objects/items/stacks/stack.dm:177]: Spam when you build

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by SplinterGP** [comment]
Just nerf the buff you did, instead of from 10 to 40 per skill just make 20%

**Comment by DerrickChamberlain** [comment]
> Does this affect the FOB drone?

It does not affect drone. I am pretty sure it does not since it is only used before shutters.

**Comment by Hydrinator** [comment]
This is a horrible idea in so many ways. Here's 3 reasons:

1. Any remotely long FOB siege is the result of uncoordinated and/or inexperienced xenos. A simple crusher + spitter combo can literally break through 4-5+ barricades protected with razorwire in seconds. Lots of turrets? Wraith. Lots of cade hugging marines? Boiler & Warrior/Ravager. Even carriers can counter these now by throwing acid huggers.

2. This would artificially create a new meta where engineers either build outside whatever this magical "build slower zone" is, build nowhere but at miners or abandon FOB building all together.

3. There is already a huge lack of engineer players. Changes like these would literally be kicking them in the balls for all the wrong reasons.

**Comment by BraveMole** [comment]
> 1. Any remotely long FOB siege is the result of uncoordinated and/or inexperienced xenos. A simple crusher + spitter combo can literally break through 4-5+ barricades protected with razorwire in seconds. Lots of turrets? Wraith. Lots of cade hugging marines? Boiler & Warrior/Ravager. Even carriers can counter these now by throwing acid huggers.enbt

Even with sweaty xenox, an excellent composition and etc, it's still a pain to attack FOB even in a 1:1 ratio. You can do it, but it takes 20 minutes of unfun gameplay


> 2. This would artificially create a new meta where engineers either build outside whatever this magical "build slower zone" is, build nowhere but at miners or abandon FOB building all together.

Good. That's the purpose of the PR. Make them do other stuff than fobitting. I agree on the artificial part though


> 3. There is already a huge lack of engineer players. Changes like these would literally be kicking them in the balls for all the wrong reasons.

I buffed the engineers tremondously in the past month. This PR barely affect one part of their gameplay, so I don't think it's right to complain about that. 

What you don't understant, is that if no one find a solution to the hard fobbit, whether by this PR or another idea, the engie buffs will all be reverted. 
Because these cancer FOB cannot stay

**Comment by Hydrinator** [comment]
> Even with sweaty xenox, an excellent composition and etc, it's still a pain to attack FOB even in a 1:1 ratio. You can do it, but it takes 20 minutes of unfun gameplay

... No, like i literally just said, it takes seconds to create breaches by utilizing a non-retarded crusher and a spitter to yeet both barricades and razorwire instantly. What takes time is killing all the marines inside if they're properly defending which these changes will not help with due to most breaches getting sealed with razorwire grenades.

> Good. That's the purpose of the PR. Make them do other stuff than fobitting. I agree on the artificial part though

Most engineers already do not give two shits about the FOB. The only ones that do are usually IRL busy or are doing FOB chores such as manning the mortar. I've had games where there are 8+ engineers and none of them don't bother putting up a single cade at the FOB, even if there's holes.

> I buffed the engineers tremondously in the past month. These buffs are allowing these utter cancer fobs. This PR barely affect one part of their gameplay, so I don't think it's right to complain about that.

Making things take slightly less time to build does not equal more engineer players. I myself didn't even notice the changes until someone told me and nor have we seen an increase in engineer players since then.

> What you don't understant, is that if no one find a solution to the hard fobbit, whether by this PR or another idea, the engie buffs will all be reverted because these cancer FOB cannot stay

Instead of looking at the FOB defenses, you can instead observe what causes cadehugging in the first place, which is currently the silo game mode that is in the process of being replaced.

Furthermore, adding more building time won't stop "cancer FOB" bases being built. It'll simply take more time to make.

**Comment by BraveMole** [comment]
> Instead of looking at the FOB defenses, you can instead observe what causes cadehugging in the first place, which is currently the silo game mode that is in the process of being replaced.

Yeah uh, hilarious? Have you ever played during old distress to say this?

**Comment by JackTheJackhammer** [comment]
The thing is, this with the worm would make FOB defense IMPOSSIBLE. I see xeno powercreep coming on and I, as a avid marine player, fear for the future of my chosen role. I have seen a queen literally walk up to a cade line, 3 marines looking at it, sitting there and placing  down a goddamn strong acid. SITTING THERE, TANKING THEIR BULLETS. This would make the problem even worse.

**Comment by BraveMole** [comment]
Not feasable without shit code or being a pain in the ass

**Pull request closed without merging**